src/common/xo/create-network-modal/index.js

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react'
+import { injectIntl } from 'react-intl'
+
+import SingleLineRow from '../../single-line-row'
+import _, { messages } from '../../messages'
+import { SelectPif } from '../../select-objects'
+import { Col } from '../../grid'
+import { createSelector } from 'selectors'
+
+class CreateNetworkModalBody extends Component {
+  _getPifPredicate = createSelector(
+    () => this.props.container,
+    container => pif => pif.$host === (container.type === 'pool' ? container.master : container.id) && pif.vlan === -1
```

Comment:
The real dependency isn't the container, it's either the host's or the pool master's id.